src/wallet/spend.cpp

```diff
@@ -1392,4 +1392,384 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
 
     return true;
 }
+
+constexpr int NUM_DENIABILIZATION_OUTPUTS = 2;
+
+static unsigned int CalculateDeniabilizationTxSize(const CScript& script, CAmount value, unsigned int numTxIn)
+{
+    // Calculation based on the comments and code in GetDustThreshold and CreateTransactionInternal
+    unsigned int txOutSize = (unsigned int)GetSerializeSize(CTxOut(value, script), PROTOCOL_VERSION);
+
+    const size_t txOutCount = NUM_DENIABILIZATION_OUTPUTS;
+    unsigned int txSize = 10 + GetSizeOfCompactSize(txOutCount); // bytes for output count
+    txSize += txOutSize * txOutCount;
+
+    int witnessversion = 0;
+    std::vector<unsigned char> witnessprogram;
+    if (script.IsWitnessProgram(witnessversion, witnessprogram)) {
+        txSize += (unsigned int)(numTxIn * 67.75f + 0.5f);
+    } else {
+        txSize += numTxIn * 148;
```

Comment:
why this magic number?
is this the number of legacy and p2sh bytes per txin?
can't name some constant with that?

Reply:
Yes, these are legacy vs Segwit tx size calculation, I think I found the code in GetDustThreshold. I'll see if we have named constants for some of those.

Reply:
and/or simple one line comments could help clarify,
I feel like named constant for magic number is a better strategy than commenting,
but maintainer should take final choice :thinking: